### PR TITLE
Fix queryShard containedctx error

### PR DESCRIPTION
**Pull request by letian-jiang**
Related to #18946 

Signed-off-by: Letian Jiang <letian.jiang@zilliz.com>

### CODE REVIEW

**Comment by letian-jiang** [comment]
rerun ut

**Comment by letian-jiang** [comment]
rerun ut

File: internal/querynode/query_node.go [hunk 1]
```diff
@@ -78,15 +78,9 @@ var Params paramtable.ComponentParam
 //  `indexCoord` is a grpc client of index coordinator.
 //  `stateCode` is current statement of this query node, indicating whether it's healthy.
 type QueryNode struct {
-	queryNodeLoopCtx    context.Context
-	queryNodeLoopCancel context.CancelFunc
-
-	wg sync.WaitGroup
-
+	ctx       context.Context
```
Comment (weiliu1031) [inline internal/querynode/query_node.go:81]: why query node still need maintain a context ?

File: internal/querynode/data_sync_service.go [hunk 2]
```diff
@@ -39,6 +39,8 @@ type dataSyncService struct {
 	metaReplica  ReplicaInterface
 	tSafeReplica TSafeReplicaInterface
 	msFactory    msgstream.Factory
+
+	closeCh chan struct{}
```
Comment (weiliu1031) [inline internal/querynode/data_sync_service.go:43]: maintain a `closeChan` here, but did nothing on it ?

Comment (letian-jiang) [reply internal/querynode/data_sync_service.go]: Removed

File: internal/querynode/query_node.go (hunk 1 above)
Comment (letian-jiang) [reply internal/querynode/query_node.go:81]: This is to keep session, chunk manager, and msgstream as is.

**Comment by letian-jiang** [comment]
rerun ut

**Pull request closed without merging**